tidb-in-kubernetes/gcp-gke-tutorial.md

```diff
@@ -0,0 +1,331 @@
+
+由于缩容过程中无法确定哪个节点会被删除，因此目前不支持集群缩容。
+
+扩容过程会持续几分钟，你可以通过 `kubectl --kubeconfig credentials/kubeconfig_<cluster_name> get po -n tidb --watch` 命令来持续观察进度。
+
+例如，可以将 `tidb_count` 从 1 改为 2 来扩容 TiDB：
+
+```
+variable "tidb_count" {
+  description = "Number of TiDB nodes per availability zone"
+  default     = 2
+}
+```
+
+> **注意：**
+>
+> 增加节点数量会在每个可用区都增加节点。
+
+## 自定义
+
+你可以按需修改 `variables.tf` 文件中的默认值，例如集群名称和镜像版本等。
+
+### 自定义 GCP 资源
+
+GCP 允许 `n1-standard-1` 或者更大的实例类型挂载本地 SSD，这提供了更好的自定义特性。
+
+### 自定义 TiDB 参数配置
+
+目前，暴露出来可以自定义修改的 TiDB 参数并不多。但是你可以在部署集群之前，修改 `templates/tidb-cluster-values.yaml.tpl` 文件中的配置。如果集群已经在运行，每次修改 `templates/tidb-cluster-values.yaml.tpl` 文件中的配置都需要执行 `terraform apply`，并手动删除 Pod(s)。
+
+### 自定义节点池
+
+集群是按区域 (regional) 而非按可用区 (zonal) 创建的。也就是说，GKE 向每个可用区复制相同的节点池，以实现更高的可用性。但对于 Grafana 这样的监控服务来说，这可能就没有必要了。可以通过 `gcloud` 手动删除节点。
+
+> **注意：**
+>
+> GKE 节点池通过实例组管理。如果你通过 `gcloud compute instances delete` 删除某个节点，GKE 会自动重新创建节点并添加到集群。
+
+假如需要从监控节点池中删掉一个节点，首先输入命令：
+
+{{< copyable "shell-regular" >}}
+
+```bash
+gcloud compute instance-groups managed list | grep monitor
+```
+
+结果类似下面输出：
+
+```
+gke-my-cluster-monitor-pool-08578e18-grp  us-west1-b  zone   gke-my-cluster-monitor-pool-08578e18  0     0            gke-my-cluster-monitor-pool-08578e18  no
+gke-my-cluster-monitor-pool-7e31100f-grp  us-west1-c  zone   gke-my-cluster-monitor-pool-7e31100f  1     1            gke-my-cluster-monitor-pool-7e31100f  no
+gke-my-cluster-monitor-pool-78a961e5-grp  us-west1-a  zone   gke-my-cluster-monitor-pool-78a961e5  1     1            gke-my-cluster-monitor-pool-78a961e5  no
+```
+
+第一列是托管的实例组，第二列是所在可用区。你还需要获取实例组中的实例名字：
+
+{{< copyable "shell-regular" >}}
+
+```bash
+gcloud compute instance-groups managed list-instances <the-name-of-the-managed-instance-group> --zone <zone>
+```
+
+示例：
+
+{{< copyable "shell-regular" >}}
+
+```bash
+gcloud compute instance-groups managed list-instances gke-my-cluster-monitor-pool-08578e18-grp --zone us-west1-b
+```
+
+```
+NAME                                       ZONE        STATUS   ACTION  INSTANCE_TEMPLATE                     VERSION_NAME  LAST_ERROR
+gke-my-cluster-monitor-pool-08578e18-c7vd  us-west1-b  RUNNING  NONE    gke-my-cluster-monitor-pool-08578e18
+```
+
+现在你可以通过指定托管的实例组和实例的名称来删掉这个实例。例如：
+
+{{< copyable "shell-regular" >}}
+
+```bash
+gcloud compute instance-groups managed delete-instances gke-my-cluster-monitor-pool-08578e18-grp --instances=gke-my-cluster-monitor-pool-08578e18-c7vd --zone us-west1-b
+```
+
+## 销毁集群
+
+可以通过如下命令销毁集群：
+
+{{< copyable "shell-regular" >}}
+
+```bash
+terraform destroy
+```
+
+如果你不再需要之前的数据，执行完 `terraform destroy` 后，可通过 Google Cloud 控制台或者 `gcloud` 删除磁盘。
+
+> **注意：**
+>
+> 在执行 `terraform destroy` 过程中，可能发生错误：`Error reading Container Cluster "my-cluster": Cluster "my-cluster" has status "RECONCILING" with message""`。当 GCP 升级 kubernetes master 节点时会出现该问题。一旦问题出现，就无法删除集群，需要等待 GCP 升级结束，再次执行 `terraform destroy`。
+
+
+## 更多信息
+
+请参考 [operation guide](./operation-guide.md)。
```

Comment:
```suggestion
请参考 [operation guide](./operation-guide.md)。
```
Please change this link into an absolute link. Otherwise, it might become a dead link on our website.

Reply:
operation guide is not ready yet, remove it for now.

Reply:
okay.